cellxgene_schema_cli/cellxgene_schema/migrate.py

```diff
@@ -192,6 +194,43 @@ def color_key_is_valid(key: str, value: str) -> bool:
         if key.endswith("_colors") and not color_key_is_valid(key, value):
             del dataset.uns[key]
 
+    # Coerce Raw Matrix values to np.float32 IF 1) assay requires this constraint in the schema definition, and
+    # 2) the max value in the matrix is representable as np.float32. NOTE: These values are all validated to be integers
+    # so no concern for data precision loss.
+
+    accessibility_assays = {
+        "EFO:0007045",  # ATAC-seq
+        "EFO:0008804",  # Methyl-seq
+        "EFO:0000751",  # methylation profiling
+        "EFO:0008939",  # snmC-seq
+    }
+    ontology_checker = ontology.OntologyChecker()
+    assays = dataset.obs.assay_ontology_term_id.drop_duplicates()
+    has_rna_assay = False
+    for assay in assays:
+        term_ancestors = ontology_checker.get_term_ancestors("EFO", assay)
+        term_ancestors.add(assay)
+        # check intersection, if no matches, term is not accessibility assay and therefore we assume its RNA
+        if bool(accessibility_assays & term_ancestors) is False:
+            has_rna_assay = True
+            break
+
+    if has_rna_assay:
+        dataset = dataset.to_memory()
+        if dataset.raw:
+            raw = dataset.raw.X
+        else:
+            raw = dataset.X
```

Comment:
does this do a deep copy?

Reply:
Nope, its a view

Reply:
I verified locally that by updating this reference var, it updates the dataset